go/cmd/dolt/commands/push.go

```diff
@@ -138,6 +138,14 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
 		case err := <-errChan:
 			return handlePushError(err, usage, apr, queryist, sqlCtx)
 		case <-ctx.Done():
+			switch ctx.Err() {
+			case context.DeadlineExceeded:
+				cli.Println("timeout exceeded")
+				return 1
+			case context.Canceled:
+				cli.Println("push cancelled by force")
+				return 1
```

Comment:
There should be a default too. Don't want to fall through is you don't have a matching case.